torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1448,11 +1449,14 @@ void ProcessGroupNCCL::ncclCommWatchdog() {
           "Process group watchdog thread terminated with exception: ",
           e.what());
       LOG(ERROR) << exitMsg;
-      // TODO(whc) clean up the rethrow - why is it stored in a class var and
-      // rethrown?
-      watchDogException_ =
-          std::make_exception_ptr(C10_BUILD_ERROR(DistBackendError, exitMsg));
-      std::rethrow_exception(watchDogException_);
+      if (C10_LIKELY(rethrowCUDAErrors_) ||
```

Comment:
Instead of defining a new ENV to control the error handling to make things more complicated, could we re-use existing ones, e.g, calling SHOULD_TEAR_DOWN(asyncErrorHandling_) here

Reply:
Would it be expected that `SHOULD_TEAR_DOWN(asyncErrorHandling_)` would not rethrow CUDA errors? If so we could consider repurposing that for this case as well.

Reply:
I think asyncErrorHandling_ is supposed to handle any errors, including CUDA errors.

Reply:
One thing to check is whether there are nccl errors that we need to handle still, bc they are not raised in the main thread. Or if all nccl errors would be raised the same way as cuda errors, it's simpler logic.